modules/rostests/apitests/wscript/wsf.c

```diff
@@ -0,0 +1,166 @@
+/*
+ * PROJECT:     ReactOS API tests
+ * LICENSE:     LGPL-2.1+ (https://spdx.org/licenses/LGPL-2.1+)
+ * PURPOSE:     Test for wscript.exe
+ * COPYRIGHT:   ReactOS Team
+ */
+
+#include <apitest.h>
+#include <windows.h>
+#include <shlwapi.h>
+
+#define MYGUID "{898AC78E-BFC7-41FF-937D-EDD01E666707}"
+
+static DWORD getregdw(HKEY hKey, LPCSTR sub, LPCSTR name, DWORD *out, DWORD defval)
+{
+    DWORD size = sizeof(*out);
+    *out = 0;
+    LRESULT e = SHGetValueA(hKey, sub, name, NULL, out, &size);
+    if (e)
+        *out = defval;
+    return e;
+}
+
+static BOOL makestringfile(LPWSTR path, LPCSTR ext, LPCSTR string, const BYTE *map)
+{
+    UINT cch = GetTempPathW(MAX_PATH, path);
+    UINT16 i = 0;
+    while (++i)
+    {
+        HANDLE hFile;
+        wsprintfW(path + cch, L"~%u.%hs", i, ext ? ext : "tmp");
+        hFile = CreateFileW(path, GENERIC_WRITE, FILE_SHARE_READ, 0, CREATE_NEW, 0, NULL);
+        if (hFile != INVALID_HANDLE_VALUE)
+        {
+            BOOL succ = TRUE;
+            for (; *string && succ; ++string)
+            {
+                BYTE ch = *string;
+                DWORD j;
+                for (j = 0; map && map[j + 0]; j += 2)
+                {
+                    if (ch == map[j + 0])
+                        ch = map[j + 1];
+                }
+                succ = WriteFile(hFile, &ch, 1, &j, NULL);
+            }
+            CloseHandle(hFile);
+            return succ;
+        }
+    }
+    return FALSE;
+}
+
+static DWORD runscriptfile(LPCWSTR path, LPCWSTR engine)
+{
+    STARTUPINFOW si;
+    PROCESS_INFORMATION pi;
+    LPCWSTR exe = engine ? engine : L"wscript.exe";
+    WCHAR cmd[MAX_PATH * 2];
+    wsprintfW(cmd, L"\"%s\" //nologo \"%s\"", exe, path);
```

Comment:
```suggestion

    wsprintfW(cmd, L"\"%s\" //nologo \"%s\"", exe, path);
```
(making the code less cluttered by separating variables and executable code)

Also, since this an APItest (and not a winetest), could you instead use `_snwprintf` or `StringCchPrintfW` (or Cb) function?

Reply:
The string will always fit.